### PR TITLE
CDAP-19313 Fix HTTP migration E2E test failures

### PR DESCRIPTION
**Pull request by njbriggs**
# CDAP-19313 Fix HTTP migration E2E test failures

## Description
Running the E2E (Cypress, for now) tests on the HTTP migration reviewed 2 significant issues:
* The error handling was not fully baked
* The Hub requires special handling

I've addressed those issues and successfully run most of the E2E tests locally.

## PR Type
- [X] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-19313](https://cdap.atlassian.net/browse/CDAP-19313)

## Test Plan
Run E2E tests

## Screenshots
N/A

### CODE REVIEW

File: app/cdap/services/datasource/DataSourceHttp.ts [hunk 1]
```diff
@@ -196,7 +230,12 @@ export default class Datasource implements IDataSource {
     const excludeFromHealthCheck = !!resource.excludeFromHealthCheck;
     const generatedResource: IResource = {
       id: resource.id || uuidV4(),
-      json: resource.json === false ? false : true,
+      // Alwas set `responseType` to 'text', even for 'json' endpoints
```
Comment (seanfeld) [inline app/cdap/services/datasource/DataSourceHttp.ts]: always :)

**Review by seanfeld — APPROVED** [review]
very small nit - doesn't need to be addressed

File: app/cdap/api/market.js [hunk 2]
```diff
@@ -16,13 +16,13 @@
```
Comment (de-lan) [inline app/cdap/api/market.js:16]: Mind also updating the year? 2016 --> 2022

**Review by de-lan — APPROVED** [review]

File: app/cdap/api/market.js (hunk 2 above)
Comment (seanfeld) [reply app/cdap/api/market.js:16]: @de-lan we don't need to update the copyright header with the new year unless its a new file or completely rewritten

**Review by mrahanjam — APPROVED** [review]

**Pull request merged**